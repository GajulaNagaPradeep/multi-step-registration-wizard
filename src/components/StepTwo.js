import React from "react";
import { useForm } from "react-hook-form";
import { Form, Button } from "react-bootstrap";
import { motion } from "framer-motion";

const StepTwo = (props) => {
  const { user } = props;
  const { register, handleSubmit, errors } = useForm({
    defaultValues: {
      user_email: user.user_email,
      user_password: user.user_password,
    },
  });

  const onPrevious = () => {
    props.history.push("/first");
  };

  const onNext = (data) => {
    props.updateUser(data);
    props.history.push("/third");
  };

  return (
    <Form className="input-form">
      <motion.div
        className="col-lg-6 offset-lg-3"
        initial={{ x: "-100vw" }}
        animate={{ x: 0 }}
        transition={{ stiffness: 150 }}
      >
        <Form.Group controlId="first_name">
          <Form.Label>Email</Form.Label>
          <Form.Control
            type="email"
            name="user_email"
            placeholder="Enter your email address"
            autoComplete="off"
            ref={register({
              required: "Email is required.",
              pattern: {
                value: /^[^@ ]+@[^@ ]+\.[^@ .]{2,}$/,
                message: "Email is not valid.",
              },
            })}
            className={`${errors.user_email ? "input-error" : ""}`}
          />
          {errors.user_email && (
            <p className="errorMsg">{errors.user_email.message}</p>
          )}
        </Form.Group>

        <Form.Group controlId="password">
          <Form.Label>Password</Form.Label>
          <Form.Control
            type="password"
            name="user_password"
            placeholder="Choose a password"
            autoComplete="off"
            ref={register({
              required: "Password is required.",
              minLength: {
                value: 6,
                message: "Password should have at-least 6 characters.",
              },
            })}
            className={`${errors.user_password ? "input-error" : ""}`}
          />
          {errors.user_password && (
            <p className="errorMsg">{errors.user_password.message}</p>
          )}
        </Form.Group>
      </motion.div>
      <Button variant="primary" type="submit" onClick={onPrevious}>
        Previous
      </Button>
      <Button
        variant="primary"
        type="submit"
        onClick={handleSubmit(onNext)}
        style={{ float: "right" }}
      >
        Next
      </Button>
    </Form>
  );
};

export default StepTwo;
