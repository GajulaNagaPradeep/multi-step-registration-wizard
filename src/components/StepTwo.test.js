import React from "react";
import { shallow } from "enzyme";
import StepTwo from "./StepTwo";

const props = {
  user: {
    first_name: "name",
    last_name: "",
    user_email: "",
    user_password: "",
  },
};

let wrapped = shallow(<StepTwo {...props} />);
describe("Progress", () => {
  it("should be rendered correctly", () => {
    expect(wrapped).toMatchSnapshot();
  });
});
