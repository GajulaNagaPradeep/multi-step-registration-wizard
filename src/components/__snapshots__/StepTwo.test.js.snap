// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress should be rendered correctly 1`] = `
<Form
  className="input-form"
  inline={false}
>
  <ForwardRef(MotionComponent)
    animate={
      Object {
        "x": 0,
      }
    }
    className="col-lg-6 offset-lg-3"
    initial={
      Object {
        "x": "-100vw",
      }
    }
    transition={
      Object {
        "stiffness": 150,
      }
    }
  >
    <FormGroup
      controlId="first_name"
    >
      <FormLabel
        column={false}
        srOnly={false}
      >
        Email
      </FormLabel>
      <FormControl
        autoComplete="off"
        className=""
        name="user_email"
        placeholder="Enter your email address"
        type="email"
      />
    </FormGroup>
    <FormGroup
      controlId="password"
    >
      <FormLabel
        column={false}
        srOnly={false}
      >
        Password
      </FormLabel>
      <FormControl
        autoComplete="off"
        className=""
        name="user_password"
        placeholder="Choose a password"
        type="password"
      />
    </FormGroup>
  </ForwardRef(MotionComponent)>
  <Button
    active={false}
    disabled={false}
    onClick={[Function]}
    type="submit"
    variant="primary"
  >
    Previous
  </Button>
  <Button
    active={false}
    disabled={false}
    onClick={[Function]}
    style={
      Object {
        "float": "right",
      }
    }
    type="submit"
    variant="primary"
  >
    Next
  </Button>
</Form>
`;
